internal/terraform/node_output.go

```diff
@@ -630,9 +628,10 @@ func (n *NodeDestroyableOutput) Execute(ctx EvalContext, op walkOperation) tfdia
 	mod := state.Module(n.Addr.Module)
 	if n.Addr.Module.IsRoot() && mod != nil {
 		s := state.Lock()
-		rootOutputs := s.RootOutputValues
-		maps.Copy(rootOutputs, s.EphemeralRootOutputValues)
```

Comment:
The false assumption I had here while reviewing this LOC in the original PR was that `rootOutputs` is a new map and the `EphemeralRootOutputValues` map is being copied into _that_ new map. Instead, because a pointer to the map is being passed around, the map is copied into the underlying `RootOutputValues` map and that's how the two maps get mixed up.